### DISCUSSION TITLE
default login not working with new deployment, muiltiple retries

### DISCUSSION THREAD

**Discussion by straes** [netbox-community/netbox-docker#478] [Getting Help] [answered]
Hello.

After multiple attempts following the quick start process here:

git clone -b release https://github.com/netbox-community/netbox-docker.git
cd netbox-docker
tee docker-compose.override.yml <<EOF
version: '3.4'
services:
  netbox:
    ports:
      - 8000:8080
EOF
docker-compose pull
docker-compose up

Everything appears to come up ok, application accessible. However, the default admin credential does not work.  This is the same result after multiple attempts, and am not altering any content in the env/netbox.yml, including these two:

SUPERUSER_NAME=admin
SUPERUSER_PASSWORD=admin

Any ideas?

`
# docker-compose up
Starting netbox-docker_postgres_1      ... done
Starting netbox-docker_redis-cache_1   ... done
Starting netbox-docker_redis_1       ... done
Starting netbox-docker_netbox-worker_1 ... done
Starting netbox-docker_netbox_1        ... done
Attaching to netbox-docker_redis_1, netbox-docker_redis-cache_1, netbox-docker_postgres_1, netbox-docker_netbox-worker_1, netbox-docker_netbox_1
postgres_1       |
postgres_1       | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres_1       |
postgres_1       | 2021-03-27 02:43:05.201 UTC [1] LOG:  starting PostgreSQL 12.6 on x86_64-pc-linux-musl, compiled by gcc (Alpine 10.2.1_pre1) 10.2.1 20201203, 64-bit
postgres_1       | 2021-03-27 02:43:05.201 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres_1       | 2021-03-27 02:43:05.201 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres_1       | 2021-03-27 02:43:05.204 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres_1       | 2021-03-27 02:43:05.228 UTC [21] LOG:  database system was shut down at 2021-03-27 02:42:58 UTC
postgres_1       | 2021-03-27 02:43:05.233 UTC [1] LOG:  database system is ready to accept connections
redis-cache_1    | 1:C 27 Mar 2021 02:43:05.047 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
redis-cache_1    | 1:C 27 Mar 2021 02:43:05.047 # Redis version=6.2.1, bits=64, commit=00000000, modified=0, pid=1, just started
redis-cache_1    | 1:C 27 Mar 2021 02:43:05.047 # Configuration loaded
redis-cache_1    | 1:M 27 Mar 2021 02:43:05.048 * monotonic clock: POSIX clock_gettime
redis-cache_1    | 1:M 27 Mar 2021 02:43:05.049 * Running mode=standalone, port=6379.
redis-cache_1    | 1:M 27 Mar 2021 02:43:05.049 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
redis-cache_1    | 1:M 27 Mar 2021 02:43:05.049 # Server initialized
redis-cache_1    | 1:M 27 Mar 2021 02:43:05.049 * Loading RDB produced by version 6.2.1
redis-cache_1    | 1:M 27 Mar 2021 02:43:05.049 * RDB age 7 seconds
redis-cache_1    | 1:M 27 Mar 2021 02:43:05.049 * RDB memory usage when created 0.77 Mb
redis-cache_1    | 1:M 27 Mar 2021 02:43:05.049 * DB loaded from disk: 0.000 seconds
redis-cache_1    | 1:M 27 Mar 2021 02:43:05.049 * Ready to accept connections
redis_1          | 1:C 27 Mar 2021 02:43:04.940 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
redis_1          | 1:C 27 Mar 2021 02:43:04.940 # Redis version=6.2.1, bits=64, commit=00000000, modified=0, pid=1, just started
redis_1          | 1:C 27 Mar 2021 02:43:04.940 # Configuration loaded
redis_1          | 1:M 27 Mar 2021 02:43:04.941 * monotonic clock: POSIX clock_gettime
redis_1          | 1:M 27 Mar 2021 02:43:04.950 * Running mode=standalone, port=6379.
redis_1          | 1:M 27 Mar 2021 02:43:04.950 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
redis_1          | 1:M 27 Mar 2021 02:43:04.950 # Server initialized
redis_1          | 1:M 27 Mar 2021 02:43:04.952 * DB loaded from append only file: 0.002 seconds
redis_1          | 1:M 27 Mar 2021 02:43:04.952 * Ready to accept connections
netbox_1         | 🧬 loaded config '/etc/netbox/config/configuration.py'
netbox_1         | 🧬 loaded config '/etc/netbox/config/extra.py'
netbox_1         | Operations to perform:
netbox_1         |   Apply all migrations: admin, auth, circuits, contenttypes, dcim, extras, ipam, secrets, sessions, taggit, tenancy, users, virtualization
netbox_1         | Running migrations:
netbox_1         |   No migrations to apply.
netbox_1         | 🧬 loaded config '/etc/netbox/config/configuration.py'
netbox_1         | 🧬 loaded config '/etc/netbox/config/extra.py'
netbox_1         | 💡 Superuser Username: admin, E-Mail: admin@example.com
netbox_1         | 🧬 loaded config '/etc/netbox/config/configuration.py'
netbox_1         | 🧬 loaded config '/etc/netbox/config/extra.py'
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/000_users.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/010_groups.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/020_custom_fields.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/020_tags.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/030_regions.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/040_sites.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/050_manufacturers.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/060_device_types.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/070_rack_roles.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/075_rack_groups.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/080_racks.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/090_device_roles.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/100_platforms.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/110_tenant_groups.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/120_tenants.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/130_cluster_types.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/135_cluster_groups.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/135_clusters.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/140_clusters.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/140_devices.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/145_devices.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/150_rirs.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/160_aggregates.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/165_cluster_groups.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/175_route_targets.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/180_vrfs.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/190_prefix_vlan_roles.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/200_vlan_groups.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/210_vlans.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/220_prefixes.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/230_virtual_machines.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/240_virtualization_interfaces.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/250_dcim_interfaces.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/260_ip_addresses.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/270_primary_ips.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/280_custom_links.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/280_providers.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/290_circuit_types.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/290_webhooks.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/300_circuits.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/310_secret_roles.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/320_services.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/330_power_panels.py
netbox_1         | ▶️  Running the startup script /opt/netbox/startup_scripts/340_power_feeds.py
netbox_1         | ✅ Initialisation is done.
netbox_1         | ⏳ Waiting for control socket to be created... (1/10)
netbox_1         | 2021/03/27 02:43:15 [warn] 1#1 Unit is running unprivileged, then it cannot use arbitrary user and group.
netbox_1         | 2021/03/27 02:43:15 [info] 1#1 unit started
netbox_1         | 2021/03/27 02:43:15 [info] 16#16 discovery started
netbox_1         | 2021/03/27 02:43:15 [notice] 16#16 module: python 3.8.7 "/usr/lib/unit/modules/python3.unit.so"
netbox_1         | 2021/03/27 02:43:15 [info] 1#1 controller started
netbox_1         | 2021/03/27 02:43:15 [notice] 1#1 process 16 exited with code 0
netbox_1         | 2021/03/27 02:43:15 [info] 18#18 router started
netbox_1         | 2021/03/27 02:43:15 [info] 18#18 OpenSSL 1.1.1k  25 Mar 2021, 101010bf
netbox_1         | 2021/03/27 02:43:15 [info] 19#19 "netbox" application started
netbox_1         | ⚙️ Applying configuration from /etc/unit/nginx-unit.json
netbox_1         | ✅ Unit configuration loaded successfully
netbox_1         | 2021/03/27 02:43:17 [notice] 1#1 process 14 exited with code 0
`

**Comment by eduncan911** [comment, marked as answer]
While I haven't ran into this issue after deleting and reinstalling a few times, one thing I do is always wipe my machine and start over.  Usually because there's always a persistent container, volume, or network that just hangs around.

Warning: this will delete all docker items you may have ever installed.

I run them so often, I made an bash alias.

```
# stop everything from running first
alias docker-clean-everything='\
  docker rm -v $(docker ps -a -q -f status=exited);\
  docker rmi -f $(docker images -q);\
  docker volume rm -f $(docker volume ls | awk "{print $2}");\
  docker network rm $(docker network ls | awk "{print $2}")'
```

Once this runs, you'll have a clean and empty machine to start over.

I also recommend deleting the git repo from your machine, and pulling it down again as they may be writing temp files.

**Reply by cimnine** [reply]
Usually `docker-compose down -v` does the job pretty well. It stops and removes all the related containers and removes all volumes as well.

**Reply by straes** [reply]
While I never could get a clear root, it only ever appeared to occur on this one instance.  All other instances tested were fine.  Thank you for the input.
